packages/wasm/src/services/credential/bound-check.ts

```diff
@@ -61,60 +62,38 @@ export function applyEnforceBounds({
   proofRequest,
   provingKeyId,
   provingKey,
+  selectedCredentials,
 }: {
   builder: PresentationBuilder;
   proofRequest: ProofRequest;
+  selectedCredentials: any[];
   provingKeyId: string;
   provingKey: LegoProvingKey;
 }) {
-  let skipProvingKey = false;
-
-  proofRequest.request.input_descriptors.forEach(inputDescriptor => {
-    inputDescriptor.constraints.fields.forEach((field: Field) => {
-      const {formatMaximum, formatMinimum, format, type} = field.filter || {};
-
-      if (!formatMaximum && !formatMinimum) {
-        return;
-      }
-
-      let max;
-      let min;
-
-      if (format === 'date-time' || format === 'date') {
-        max = new Date(formatMaximum || MAX_DATE_PLACEHOLDER);
-        min = new Date(formatMinimum || MIN_DATE_PLACEHOLDER);
-      } else if (type === 'number') {
-        max = formatMaximum || MAX_NUMBER;
-        min = formatMinimum || MIN_INTEGER;
-      } else {
-        throw new Error(
-          `Unsupported format ${format} and type ${type} for enforce bounds`
-        )
-      }
-
-      const attributeName = field.path.join('.').replace('$.', '');
-
-      console.log('enforceBounds', {
-        attributeName,
-        min,
-        max,
-      });
-
-      builder.enforceBounds(
-        0,
-        attributeName,
-        min,
-        max,
-        provingKeyId,
-        !skipProvingKey ? provingKey : undefined,
-      );
-
-      // Proving key will be used only for the first enforce bounds call
-      skipProvingKey = true;
-    });
+  const descriptorBounds = pexToBounds(
+    proofRequest.request,
+    selectedCredentials,
+  );
+  let bounds = [];
+
+  descriptorBounds.forEach((items) => {
+    items.forEach((bound, idx) => {
```

Comment:
this is incorrect, the pextobounds result is split per credential. this wont work with multiple credentials.

descriptorBounds[0] corresponds to the first selected credential
descriptorBounds[1] corresponds to the second selected credential

Reply:
yes, i've just pushed an update to support multiple credentials

Reply:
still finishing some tests